src/Components/AddIssue/index.tsx

```diff
@@ -1,71 +1,139 @@
 import React, { ChangeEvent } from 'react';
-import { IIssue } from '../../Domain/Issues';
+import { IIssue, Issue } from '../../Domain/Issues';
 import './index.css';
+import issueTypesData from "../../JsonFiles/IssueType.json"
+import {IIssueType} from "../../Domain/IssueType"
+import { IIssueLocation } from "../../Domain/IssueLocation";
+import {IIssueLocationGateway as ILoadIssueLocationGateway} from '../../Boundary/IssueLocation/'
+import { IServiceProvider, ServiceContext } from '../../ServiceContext';
+import { v4 as uuid } from 'uuid';
 
 export interface IAddIssuesProps
 {
-  onChangeIssue: (issue: IIssue, index: number) => void;
-  onDeleteIssue: (index:number) => void;
+  onChangeIssue: (issues: IIssue, index: number) => void;
+  onDeleteIssue: (index: number) => void;
+  issue:IIssue;
   index: number;
-  issue: IIssue;
 }
 
 export interface IAddIssueState
 {
   issue:IIssue;
+  issueTypes: IIssueType[];
+  issueLocations: IIssueLocation[];
 }
 
 export class AddIssue extends React.Component<IAddIssuesProps,IAddIssueState> {
+    public static contextType = ServiceContext;
+
+    private _issueTypes = Array.from<IIssueType>(issueTypesData);
+    private _issueLocations = new Array<IIssueLocation>();
+    private _issueLoadLocationGateway: ILoadIssueLocationGateway;
 
-    constructor(props:IAddIssuesProps)
+    public constructor(props:IAddIssuesProps, context: IServiceProvider)
     {
       super(props);
-
+      this._issueLoadLocationGateway = context.get<ILoadIssueLocationGateway>("ILoadIssueLocationGateway");
+      this.loadIssueLocations();
       this.state={
-        issue: this.props.issue
+        issue: this.props.issue,
+        issueTypes: this._issueTypes,
+        issueLocations: this._issueLocations,
       }
     }
-    deleteIssue(){
+
+    loadIssueLocations() {
+      this._issueLoadLocationGateway.loadIssueLocations().then((data) => {
+        this._issueLocations = data.issueLocations;
+        this.setState({ issueLocations: data.issueLocations });
+        let issue = this.state.issue;
+        issue.Location = this.state.issueLocations[0];
+        this.setState({ issue: issue });
+      });
+    }
+
+    deleteIssue = (): void => {
       this.props.onDeleteIssue(this.props.index);
     }
 
-    handleChangeOfIssue = (event: ChangeEvent<HTMLInputElement>): void => {
-      const name = event.target.name;
+    handleChangeOfIssueNote=(event: ChangeEvent<HTMLTextAreaElement>): void => {
       const value = event.target.value;
-      this.setState({
-          issue: { 
-              ...this.state.issue,
-              [name]: [value]
-          }
-      });
+
+      let issue = this.state.issue;
+      issue.Notes = value;
+      this.setState({issue:issue});
+
+      this.props.onChangeIssue(this.state.issue, this.props.index);
+      console.log(this.state.issue);
```

Comment:
Probably remove all the console.log statements

Reply:
Yep agreed.

Reply:
Resolved in 
49ae60f